### PR TITLE
Upgrade fluentd base image, refactor docker file to pull steps into builder image, remove dependencies that are not needed.

### PR DESCRIPTION
**Pull request by frankreno**
Upgrade fluentd base image, refactor docker file to pull steps into builder image, remove dependencies that are not needed.

### CODE REVIEW

File: deploy/docker/Dockerfile [hunk 1]
```diff
@@ -67,10 +56,19 @@ RUN mkdir /tmp/terraform && \
     chmod +x /bin/terraform && \
     rm -rf /tmp/terraform
 
+# Start with fresh image
+FROM fluent/fluentd:v1.11.1-debian-1.0
+
+USER root
```
Comment (perk-sumo) [inline deploy/docker/Dockerfile:62]: I wonder if we could drop the `root` user afer all of the apt-gets, mkdirs and chowns.

**Comment by frankreno** [comment]
Interesting. The build script passes locally. I'll take a look.

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (frankreno) [reply deploy/docker/Dockerfile:62]: We don't run as root, we run the image as the fluent user.

**Comment by frankreno** [comment]
CI Is passing now after removing tracing components.  @sumo-drosiek @perk-sumo please confirm I did not miss anything in the removal.

File: tests/tracing/static/collection-monitoring-false.input.yaml [hunk 2]
```diff
@@ -1,6 +0,0 @@
-sumologic:
```
Comment (sumo-drosiek) [inline tests/tracing/static/collection-monitoring-false.input.yaml:1]: This is test for otelcol, we shouldn't remove it

File: tests/tracing/static/simple.output.yaml [hunk 3]
```diff
@@ -1,114 +0,0 @@
...
-kind: ConfigMap
-metadata:
-  name: RELEASE-NAME-sumologic-otelcol
-  labels:
-    app: RELEASE-NAME-sumologic-fluentd-metrics
-    chart: "sumologic-1.1.0-beta.2"
-    release: "RELEASE-NAME"
-    heritage: "Helm"
-data:
-  traces.otelcol.conf.yaml: |-
```
Comment (sumo-drosiek) [inline tests/tracing/static/simple.output.yaml:13]: This is test for otelcol, we shouldn't remove it

Comment (frankreno) [reply tests/tracing/static/simple.output.yaml:13]: But these tests reference the configmap, which is now removed?

Comment (sumo-drosiek) [reply tests/tracing/static/simple.output.yaml:13]: it references otelcol configmap

Comment (frankreno) [reply tests/tracing/static/simple.output.yaml]: Ahh, got it.  Ok fixed this.

**Comment by frankreno** [comment]
Closing in favor of #817

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (perk-sumo) [reply deploy/docker/Dockerfile:61]: Ah, I see :) 
https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/775/files#diff-aa72c1ddeec59a31f30834d956165473R78

**Pull request closed without merging**